contrib/epee/src/mlog.cpp

```diff
@@ -134,9 +134,58 @@ void mlog_configure(const std::string &filename_base, bool console, const std::s
   el::Loggers::addFlag(el::LoggingFlag::DisableApplicationAbortOnFatalLog);
   el::Loggers::addFlag(el::LoggingFlag::ColoredTerminalOutput);
   el::Loggers::addFlag(el::LoggingFlag::StrictLogFileSizeCheck);
-  el::Helpers::installPreRollOutCallback([filename_base](const char *name, size_t){
+  el::Helpers::installPreRollOutCallback([filename_base, max_log_files](const char *name, size_t){
     std::string rname = generate_log_filename(filename_base.c_str());
     rename(name, rname.c_str());
+    if (max_log_files != 0)
+    {
+      std::vector<boost::filesystem::path> found_files;
+      const boost::filesystem::directory_iterator end_itr;
+      for (boost::filesystem::directory_iterator iter(boost::filesystem::path(filename_base).parent_path()); iter != end_itr; ++iter)
+      {
+        const std::string filename = iter->path().string();
+        if (filename.size() >= filename_base.size() && std::memcmp(filename.data(), filename_base.data(), filename_base.size()) == 0)
```

Comment:
That's also boost::starts_with(filename, filename_base) fwiw (no need to change)

Reply:
Right, I'll try to remember it.